### PR TITLE
📝 Parameter and ParameterFrame documentation

### PR DESCRIPTION
**Pull request by je-cook**
## Linked Issues

Part of #1426 

## Description

Updated documentation for Parameter and ParameterFrame

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Review by hsaunders1904 — COMMENTED** [review]
Looks good, just a few of comments.

I think it may be a good idea to shout about the unit conversions a little more. Maybe we could dedicate a whole subheading (under `ParameterFrames`) to explain what's going on, rather than a single sentence.

Maybe something along the lines of:

> Units
> -----
> `ParameterFrame`s always enforce the same set of standard units \<link to conventions\>. 
> Parameters within a ParameterFrame whose units are convertible to one of bluemira's standard units, have their values and converted to the corresponding standard unit. This keeps the units used within Bluemira consistent across classes and modules. For this reason, if your inputs use a non-standard unit, the value you put into a `Parameter` will be different to the one you get out. 
>
> \<code example\>
>
> Use `Parameter.value_as` to return parameter values in a non-standard unit
> Input values with units not listed in \<link to unit conventions\> are not modified.

File: documentation/source/base/parameter.rst [hunk 1]
```diff
@@ -1,290 +1,113 @@
...
 :py:class:`Parameter` and :py:class:`ParameterFrame` are the means by which the physical
 configuration of a ``bluemira`` analysis are setup.
 
-The Parameter class
--------------------
+Parameters
+^^^^^^^^^^
 
-Parameters provide the value as well as a set of supporting metadata, including:
+A :py:class:`Parameter` is a dataclass that wraps a value acting a container for metadata about the value. To create a :py:class:`Parameter` as a minimum the name, the value and a unit need to be provided but other information can be added:
```
[the remark this answers is not in the document]
Comment (je-cook) [reply documentation/source/base/parameter.rst]: Covered

[the remark this answers is not in the document]
Comment (je-cook) [reply documentation/source/base/parameter.rst]: covered

File: documentation/source/base/parameter.rst [hunk 2]
```diff
@@ -112,3 +118,39 @@ The type of each :py:class:`Parameter` must be specified and adhered to in the i
...
+    ... ╒════════╤═════════╤═══════════════╤══════════╤═══════════════╤═════════════╕
+    ... │ name   │   value │ unit          │ source   │ description   │ long_name   │
+    ... ╞════════╪═════════╪═══════════════╪══════════╪═══════════════╪═════════════╡
+    ... │ A      │     3.1 │ dimensionless │ Input    │ N/A           │ N/A         │
+    ... ├────────┼─────────┼───────────────┼──────────┼───────────────┼─────────────┤
+    ... │ R_0    │    0.09 │ m             │ Input    │ N/A           │ N/A         │
+    ... ╘════════╧═════════╧═══════════════╧══════════╧═══════════════╧═════════════╛
+
+Use :py:meth:`Parameter.value_as` to return parameter values in a non-standard unit
+Input values with units listed in :ref:`unit_convention` are not modified.
```
Comment (hsaunders1904) [inline documentation/source/base/parameter.rst:156]:
```suggestion
Input values with units not listed in :ref:`unit_convention` are not modified.
``

File: documentation/source/base/parameter.rst [hunk 3]
```diff
@@ -7,7 +7,9 @@ configuration of a ``bluemira`` analysis are setup.
 Parameters
 ^^^^^^^^^^
 
-A :py:class:`Parameter` is a dataclass that wraps a value acting a container for metadata about the value. To create a :py:class:`Parameter` as a minimum the name, the value and a unit need to be provided but other information can be added:
+A :py:class:`Parameter` is a class that wraps a value acting a container for metadata about the value.
```
Comment (hsaunders1904) [inline documentation/source/base/parameter.rst:10]:
```suggestion
A :py:class:`Parameter` object wraps a value and holds metadata relating to that value.
```

**Review by hsaunders1904 — APPROVED** [review]
Last couple of nitpicks, but I think this looks really good!

**Comment by hsaunders1904** [comment]
Oh, and I'm not sure if you spotted this comment: https://github.com/Fusion-Power-Plant-Framework/bluemira/pull/1542#discussion_r1004340865

File: documentation/source/base/parameter.rst (hunk 2 above)
Comment (hsaunders1904) [reply documentation/source/base/parameter.rst:156]: Actually, maybe we can do better here:

```suggestion
Input values with units not convertible to those listed in :ref:`unit_convention` are not modified.
```

Comment (je-cook) [reply documentation/source/base/parameter.rst:157]: Discussed elsewhere

**Pull request merged**